register.py

```diff
@@ -125,120 +130,130 @@ def generate_otp(self):
-                        secondAdress="",
-                        city="",
-                        province="",
-                        country="",
-                        postalCode="",
-                        gender="",
-                        religion="",
-                        verification="",
-                        verification_material="",
-                        username=username,
-                        password=password,
-                        password_confirmation=password_confirmation,
-                        verification_id=verification_method,
-                        verification_method=verification,
-                    )
+                # handle appropriate validated_registration -> check if the username and password already existed
+                if len(validated_registration) == 4 and len(register_errors) == 0:
+                    # genrate the salt for the hashed_password
+                    salt = bcrypt.gensalt()
+                    # hash the password using bcrypt hashing algorithm
+                    hashed_password = bcrypt.hashpw(password.encode("utf-8"), salt)
 
-                    # handle appropriate validated_registration
-                    if len(validated_registration) == 4 and len(register_errors) == 0:
-                        # genrate the salt for the hashed_password
-                        salt = bcrypt.gensalt()
-                        # hash the password using bcrypt hashing algorithm
-                        hashed_password = bcrypt.hashpw(password.encode("utf-8"), salt)
+                    decoded_hashed_password = hashed_password.decode("utf-8")
 
-                        decoded_hashed_password = hashed_password.decode("utf-8")
+                    # query each field to make sure each of them is unique
+                    username_taken = (
+                        Users.query.filter(Users.username == username).first()
+                        is not None
+                    )
 
-                        # query each field to make sure each of them is unique
-                        username_taken = (
-                            Users.query.filter(Users.username == username).first()
-                            is not None
-                        )
+                    password_taken = (
+                        Users.query.filter(
+                            Users.password == decoded_hashed_password
+                        ).first()
+                        is not None
+                    )
 
-                        password_taken = (
-                            Users.query.filter(
-                                Users.password == decoded_hashed_password
-                            ).first()
-                            is not None
-                        )
+                    verification_method_taken = (
+                        Users.query.filter(
+                            Users.verification_method == verification_method,
```

Comment:
This is possibly a bug, though I'm not 100% sure. `Users.vertification_method` is a static field, it might not return what you expect.

So from reading the code, the goal is to check for uniqueness of user. Why is this check for `verification_method` is needed? There can be more than 1 users using the same verification method: email for example.

We only need to check for uniqueness of username, email, and password.

Reply:
On a different note, is there a reason why we want to store both `username` and `email`? I'm thinking to only store `email`.

Big change though so don't worry about it in this PR.